test/elixir_analyzer/exercise_test/common_checks/variable_names_test.exs

```diff
@@ -0,0 +1,419 @@
+# credo:disable-for-this-file Credo.Check.Readability.VariableNames
+
+defmodule ElixirAnalyzer.ExerciseTest.CommonChecks.VariableNamesTest do
+  use ExUnit.Case
+  alias ElixirAnalyzer.Comment
+  alias ElixirAnalyzer.Constants
+  alias ElixirAnalyzer.ExerciseTest.CommonChecks.VariableNames
+
+  @moduledoc """
+  Tests that variables are in snake_case
+  tests taken and adapted from
+  https://github.com/rrrene/credo/blob/master/test/credo/check/readability/variable_names_test.exs
+  """
+
+  describe "valid variable names" do
+    test "it should NOT report expected code" do
+      code =
+        quote do
+          defmodule CredoSampleModule do
+            def some_function(parameter1, parameter2) do
+              some_value = parameter1 + parameter2
+              {some_value, _} = parameter1
+              [1, some_value] = parameter1
+              [some_value | tail] = parameter1
+              "e" <> some_value = parameter1
+              ^some_value = parameter1
+              %{some_value: some_value} = parameter1
+              #    ... = parameter1 # Ignore this pathological case
```

Comment:
Since this is a single variable name that's so weird, let's add a case for it and make this pass :D it should be trivial, and I would want to avoid false-positives. Meanwhile I won't be able to sleep until I find out what is going on, why is this a valid variable name...